# InPhyNet

[InPhyNet](https://github.com/NathanKolbow/InPhyNet.jl) is a [Julia](https://julialang.org/) package for constructing large phylogenetic networks from a set of smaller, disjoint networks.

### Why use InPhyNet?

InPhyNet enables the inference of large-scale phylogenetic networks that would otherwise be computationally infeasible to infer with methods like [SNaQ](https://github.com/JuliaPhylo/SNaQ.jl), [PhyNEST](https://github.com/sungsik-kong/PhyNEST.jl), or [PhyloNet](https://phylogenomics.rice.edu/html/phylonetTutorial.html). Additionally, the InPhyNet.jl Julia package is easy to install and run if you already have estimated gene trees or several small phylogenetic networks that you want to merge.

### How does InPhyNet work?

InPhyNet is a heuristics-based method that constructs a species network using a distance matrix $D$ and a set of constraint networks $\mathbf{N}$.

### Citation

```tex
pending BibTeX citation
```
